Source/Immutable/Public/Immutable/Marketplace/ImtblSwapRequest.h

```diff
@@ -0,0 +1,77 @@
+﻿#pragma once
+
+#include "ImtblSwapRequest.generated.h"
+
+enum class EImtblEnvironment : uint8;
+
+/**
+ * Swap request query params
+ */
+USTRUCT(BlueprintType)
+struct FImtblSwapRequestQueryParams
+{
+	GENERATED_BODY()
+
+public:
+	FString GetPercentEncodedUrl() const;
+
+public:
+	UPROPERTY(EditAnywhere, BlueprintReadWrite, Category = "Immutable")
+	FString PublishableKey;
+
+	UPROPERTY(EditAnywhere, BlueprintReadWrite, Category = "Immutable")
+	FString FromTokenAddress;
+
+	UPROPERTY(EditAnywhere, BlueprintReadWrite, Category = "Immutable")
+	FString ToTokenAddress;
+};
+
+/**
+ * Functionality to generate swap requests
+ */
+// UCLASS(BlueprintType, Config = "Immutable", DefaultConfig)
+UCLASS(BlueprintType, Config = "Immutable")
+class IMMUTABLE_API UImtblSwapRequest : public UObject
+{
+	GENERATED_BODY()
+
+public:
+	/** 
+	 * Finds the base url associated with the environment.
+	 *
+	 * @param	Environment		The environment that will be used to look the value up.
+	 * @param	BaseUrl			The base url associated with the key.
+	 * @return					True if an item was found (False indicates nothing in the map uses the provided environment).
+	 */
+	UFUNCTION(BlueprintPure, Category = "Immutable")
+	bool GetBaseUrl(EImtblEnvironment Environment, FString& BaseUrl) const;
```

Comment:
Swap only works on sandbox and product. Can we make that clear.
Same for GetApiKey